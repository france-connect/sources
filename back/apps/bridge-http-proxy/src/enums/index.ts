// Stryker disable all
/* istanbul ignore file */

// Declarative code
export * from './bridge-http-proxy-routes.enum';
export * from './error-code.enum';
