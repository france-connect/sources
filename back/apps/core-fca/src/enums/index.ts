/* istanbul ignore file */

// Declarative file
export * from './error-code.enum';
export * from './event-categories.enum';
