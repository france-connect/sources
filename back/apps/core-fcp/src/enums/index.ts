// Stryker disable all
/* istanbul ignore file */

// Declarative code
export * from './emails-templates.enum';
export * from './error-code.enum';
export * from './process-core.enum';
