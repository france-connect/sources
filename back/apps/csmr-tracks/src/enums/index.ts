/* istanbul ignore file */

// Declarative code
export * from './csmr-tracks-error-code.enum';
export * from './instances.enum';
export * from './platform.enum';
