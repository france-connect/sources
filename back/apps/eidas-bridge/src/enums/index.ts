/* istanbul ignore file */

// Declarative code
export * from './eidas-bridge-routes.enum';
export * from './error-code.enum';
export * from './events-categories.enum';
