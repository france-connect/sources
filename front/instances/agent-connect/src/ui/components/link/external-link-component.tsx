import classnames from 'classnames';
import React from 'react';

import { LinkComponentProps, Sizes } from '@fc/dsfr';

export interface ExternalLinkComponentProps extends LinkComponentProps {
  className?: string;
  href: string;
  icon?: string;
  iconPlacement?: 'right' | 'left';
  label?: string;
  size?: Sizes;
  target?: '_blank';
  rel?: string;
}

export const ExternalLinkComponent: React.FC<ExternalLinkComponentProps> = React.memo(
  ({
    className,
    href,
    icon,
    iconPlacement,
    label,
    rel,
    size,
    target,
  }: ExternalLinkComponentProps) => (
    // @TODO add an URL validators
    // it will be created with any backoffice app
    <a
      className={classnames(
        `fr-link fr-link--${size}`,
        {
          [`fr-icon-${icon}`]: !!icon,
          [`fr-link--icon-${iconPlacement}`]: !!icon,
        },
        className,
      )}
      href={href}
      rel={rel}
      target={target}>
      {label}
    </a>
  ),
);

ExternalLinkComponent.defaultProps = {
  className: undefined,
  icon: undefined,
  iconPlacement: 'left',
  label: undefined,
  rel: 'noopener noreferrer',
  size: Sizes.MEDIUM,
  target: '_blank',
};

ExternalLinkComponent.displayName = 'ExternalLinkComponent';
