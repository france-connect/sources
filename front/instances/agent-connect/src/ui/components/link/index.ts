/* istanbul ignore file */

// declarative file
export * from './external-link-component';
