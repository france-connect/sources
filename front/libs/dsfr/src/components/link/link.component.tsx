import classnames from 'classnames';
import React from 'react';

import { Sizes } from '../../enums';

export interface LinkComponentProps {
  className?: string;
  href: string;
  icon?: string;
  iconPlacement?: 'right' | 'left';
  label?: string;
  size?: Sizes;
}

export const LinkComponent: React.FC<LinkComponentProps> = React.memo(
  ({ className, href, icon, iconPlacement, label, size }: LinkComponentProps) => (
    // @TODO add an URL validators
    // it will be created with any backoffice app
    <a
      className={classnames(
        `fr-link fr-link--${size}`,
        {
          [`fr-icon-${icon}`]: !!icon,
          [`fr-link--icon-${iconPlacement}`]: !!icon,
        },
        className,
      )}
      href={href}>
      {label}
    </a>
  ),
);

LinkComponent.defaultProps = {
  className: undefined,
  icon: undefined,
  iconPlacement: 'left',
  label: undefined,
  size: Sizes.MEDIUM,
};

LinkComponent.displayName = 'LinkComponent';
