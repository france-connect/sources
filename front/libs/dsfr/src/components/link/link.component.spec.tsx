import { render } from '@testing-library/react';

import { Sizes } from '../../enums';
import { LinkComponent } from './link.component';

describe('LinkComponent', () => {
  it('should match the snapshot, with default props', () => {
    // when
    const { container } = render(<LinkComponent href="any-url-mock" />);

    // then
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot, with size sm props', () => {
    // when
    const { container } = render(<LinkComponent href="any-url-mock" size={Sizes.SMALL} />);

    // then
    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveClass('fr-link fr-link--sm');
  });

  it('should match the snapshot, with size lg props', () => {
    // when
    const { container } = render(<LinkComponent href="any-url-mock" size={Sizes.LARGE} />);

    // then
    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveClass('fr-link fr-link--lg');
  });

  it('should match the snapshot, with icon props', () => {
    // when
    const { container } = render(<LinkComponent href="any-url-mock" icon="any-icon-mock" />);

    // then
    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveClass('fr-icon-any-icon-mock');
  });

  it('should match the snapshot, with iconPlacement props', () => {
    // when
    const { container } = render(
      <LinkComponent href="any-url-mock" icon="any-icon-mock" iconPlacement="right" />,
    );

    // then
    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveClass('fr-link--icon-right');
  });

  it('should match the snapshot, with label props', () => {
    // when
    const { container, getByText } = render(
      <LinkComponent href="any-url-mock" label="any-label-mock" />,
    );
    const element = getByText('any-label-mock');

    // then
    expect(container).toMatchSnapshot();
    expect(element).toBeInTheDocument();
  });

  it('should match the snapshot, with className props', () => {
    // when
    const { container } = render(
      <LinkComponent className="any-className-mock" href="any-url-mock" />,
    );

    // then
    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveClass('any-className-mock');
  });
});
