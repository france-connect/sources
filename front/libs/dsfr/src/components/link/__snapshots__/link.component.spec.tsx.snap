// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkComponent should match the snapshot, with className props 1`] = `
<div>
  <a
    class="fr-link fr-link--md any-className-mock"
    href="any-url-mock"
  />
</div>
`;

exports[`LinkComponent should match the snapshot, with default props 1`] = `
<div>
  <a
    class="fr-link fr-link--md"
    href="any-url-mock"
  />
</div>
`;

exports[`LinkComponent should match the snapshot, with icon props 1`] = `
<div>
  <a
    class="fr-link fr-link--md fr-icon-any-icon-mock fr-link--icon-left"
    href="any-url-mock"
  />
</div>
`;

exports[`LinkComponent should match the snapshot, with iconPlacement props 1`] = `
<div>
  <a
    class="fr-link fr-link--md fr-icon-any-icon-mock fr-link--icon-right"
    href="any-url-mock"
  />
</div>
`;

exports[`LinkComponent should match the snapshot, with label props 1`] = `
<div>
  <a
    class="fr-link fr-link--md"
    href="any-url-mock"
  >
    any-label-mock
  </a>
</div>
`;

exports[`LinkComponent should match the snapshot, with size lg props 1`] = `
<div>
  <a
    class="fr-link fr-link--lg"
    href="any-url-mock"
  />
</div>
`;

exports[`LinkComponent should match the snapshot, with size sm props 1`] = `
<div>
  <a
    class="fr-link fr-link--sm"
    href="any-url-mock"
  />
</div>
`;
